src/routes/(app)/settings/+page.server.ts

```diff
@@ -44,46 +38,56 @@ export const actions: Actions = {
     const { user, prisma } = locals;
     const form = await request.formData();
     if (!user) return fail(401, { form });
-    const subscription: FormDataEntryValue[] = form.getAll("subscription");
-    const push: FormDataEntryValue[] = form.getAll("push");
-    const tags: FormDataEntryValue[] = form.getAll("tag");
 
-    // Try-catch if for some reason form data isn't correct
-    try {
-      await prisma.subscriptionSetting.deleteMany({
-        where: {
-          memberId: user.memberId,
-        },
-      });
-      await prisma.subscriptionSetting.createMany({
-        data: subscription.map((sub) => {
-          return {
-            memberId: user.memberId as string,
-            type: sub.toString(),
-            pushNotification: push.find(
-              (tag) => sub.toString() == tag.toString(),
-            )
-              ? true
-              : false,
-          };
-        }),
-      });
-      await prisma.member.update({
-        where: {
-          id: user.memberId as string,
-        },
-        data: {
-          subscribedTags: {
-            set: tags.map((tag) => {
-              return {
-                id: tag.toString(),
-              };
-            }),
-          },
-        },
-      });
-    } catch {
-      return fail(400, { form });
+    // Some error handling for the form data
+    // The type received is FormDataEntryValue, which is string | File
+    // Only string should be possible, but we make TS happy this way
+    // And it doesn't hurt to check that the values are strings
+    const push: string[] = [];
+    for (const v of form.getAll("push")) {
+      if (typeof v !== "string") return fail(400, { form });
+      push.push(v);
+    }
+    const subscription: string[] = [];
+    for (const v of form.getAll("subscription")) {
+      if (typeof v !== "string") return fail(400, { form });
+      subscription.push(v);
+    }
+    const tags: string[] = [];
+    for (const v of form.getAll("tag")) {
+      if (typeof v !== "string") return fail(400, { form });
+      tags.push(v);
     }
+
+    const memberId = user.memberId;
+    if (memberId == null) return fail(400, { form });
+    await prisma.subscriptionSetting.deleteMany({
+      where: {
+        memberId: memberId,
+      },
+    });
+    await prisma.subscriptionSetting.createMany({
+      data: subscription.map((sub) => {
+        return {
+          memberId: memberId,
+          type: sub,
+          pushNotification: push.find((tag) => sub == tag) ? true : false,
+        };
+      }),
+    });
+    await prisma.member.update({
+      where: {
+        id: memberId,
+      },
+      data: {
+        subscribedTags: {
+          set: tags.map((tag) => {
+            return {
+              id: tag,
+            };
+          }),
+        },
+      },
+    });
```

Comment:
Should probably be done in a transaction in-case one of these fails. Because in that case a user might lose all their subscription settings but none would be added or something. I know this isn't written in this PR but always good to improve our code.

Also, we don't have to remove all of the user's settings, only remove the ones not in the form data.